Client/Frontend/Sync/BraveCore/Bookmarks/Bookmarkv2.swift

```diff
@@ -309,24 +359,28 @@ extension Bookmarkv2 {
             }
         }
         
-        self.observer = Bookmarkv2.bookmarksAPI.add(observer)
+        self.bookmarkFavIconObserver = bookmarksAPI.add(observer)
     }
     
     public func removeFavIconObserver() {
-        observer = nil
+        bookmarkFavIconObserver = nil
     }
     
     public static func waitForBookmarkModelLoaded(_ completion: @escaping () -> Void) {
+        guard let bookmarksAPI = Bookmarkv2.bookmarksAPI else {
+            return
+        }
+        
         if bookmarksAPI.isLoaded {
             DispatchQueue.main.async {
                 completion()
             }
         } else {
-            var observer: BookmarkModelListener?
-            observer = Bookmarkv2.bookmarksAPI.add(BookmarkModelStateObserver({
+            //var observer: BookmarkModelListener?
```

Comment:
is this still needed?

Reply:
Fix at https://github.com/brave/brave-ios/pull/3703/commits/859375beafc1a0b1778a58dd79cb21c16bacd6cf